tests/authentication/04-check-keys-remote.t

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Checks remote ZMQ keys are created and deleted on shutdown.
+. "$(dirname "$0")/test_header"
+set_test_remote_host
+set_test_number 4
+init_suite "${TEST_NAME_BASE}" <<'__SUITE_RC__'
+[cylc]
+[scheduling]
+    [[graph]]
+        R1 = holder => held
+[runtime]
+    [[holder]]    
+        script = """cylc hold "${CYLC_SUITE_NAME}" """
+        [[[remote]]]
+            host = $CYLC_TEST_HOST
+    [[held]]
+        script = """sleep 10"""
```

Comment:
```suggestion

```

This test looks good now, but `sleep 10` is unnecessary as the `held` task will never run.